webui/react/src/services/apiConfig.ts

```diff
@@ -193,16 +193,26 @@ export const resetUserSetting: DetApi<
 export const getUserPermissions: DetApi<
   Service.GetUserParams,
   Api.V1GetPermissionsSummaryResponse,
-  Type.Permission[]
+  Type.PermissionsSummary
 > = {
   name: 'getUserPermissions',
-  postProcess: (response) => {
-    let permissions = new Array<Type.Permission>();
-    response.roles
-      .map(decoder.mapV1Role)
-      .forEach((r) => (permissions = permissions.concat(r.permissions)));
-    return permissions;
-  },
+  postProcess: (response) => ({
+    assignments: response.assignments.map(decoder.mapV1UserAssignment),
+    roles: response.roles.map(decoder.mapV1Role),
+  }),
+  request: (params) => detApi.RBAC.getPermissionsSummary(params.userId),
+};
+
+export const getPermissionsSummary: DetApi<
```

Comment:
This name doesn't make it clear what this does. Can we add a JSDoc explaining that this fetches the permissions summary for the currently logged-in user?

Reply:
Added JSDoc here and to the similar `getUserPermissions` in the latest commit; hope that helps.